src/quo2/components/channel_avatar.cljs

```diff
@@ -0,0 +1,48 @@
+(ns quo2.components.channel-avatar
+  (:require [quo2.foundations.colors :as colors]
+            [quo.react-native :as rn]
+            [status-im.ui.components.icons.icons :as icons]))
+
+(defn channel-avatar [{:keys [big? dark? lock-status icon x y]}]
+  (let [locked? (= :locked lock-status)
+        lock-exists? (not= :none lock-status)]
+    [rn/view {:style {:width (if big? 32 24)
+                      :height (if big? 32 24)
+                      :top (int x)
+                      :left (int y)
+                      :border-radius (if big? 32 24)
+                      :background-color (if dark?
+                                          colors/neutral-70
+                                          colors/neutral-30)}}
+     [rn/view {:style {:left (if big? 6 3)
+                       :top (if big? 6 3)
+                       :width 20
+                       :height 20
+                       :display :flex
+                       :justify-content :center
+                       :align-items :center}}
+      [icons/icon (keyword "main-icons" icon)
+       {:color "nil"
```

Comment:
"nil" ? could you elaborate ?

Reply:
An invalid color value would fall back to the original icon's color

Reply:
Made the color an input instead